main/webapp/modules/core/scripts/dialogs/clustering-dialog.js

```diff
@@ -512,13 +534,43 @@ ClusteringDialog.prototype._getBaseClusters = function() {
 
 ClusteringDialog.prototype._getRestrictedClusters = function(except) {
     var clusters = this._getBaseClusters();
+    let newClusters = [];
+    for (var i = 0; i < clusters.length; i++) {
+        let cluster = clusters[i];
+        let newCluster = {
+            edit: false,
+            choices: [],
+            value: cluster.value
+        };
+        
+        if (cluster.edit) {
+            for (var j = 0; j < cluster.choices.length; j++) {
+                let checkBoxID = 'Checkbox' + i.toString() + "_Choice" + j.toString();
+                let checkBox = document.getElementById(checkBoxID);
+                if(checkBox.checked){
+                    newCluster.choices.push(cluster.choices[j]);
+                    if(newCluster.choices.length == 1){
+                        newCluster.edit = true;
+                    }
+                }
+            }
+        }
+        if(Array.isArray(newCluster.choices) && newCluster.choices.length){
+            newClusters.push(newCluster)
+        }
+    }
```

Comment:
@zyadtaha in this method I am not sure why you needed to add this new code. This seems related to the bug I mentioned above, no? Could you explain me why this method needed adapting? On the surface, it feels to me like it could probably be kept as it used to be, as the checkboxes you introduced should have no influence on the restrictions via facets.

Reply:
@wetneb Yes, sorry, that was a mistake. The checkboxes have nothing related to the restrictions.
I returned the previous implementation and therefore adjusted the `ClusteringDialog.prototype._apply` function with my new code.

It works fine now. Thanks!

![Screenshot from 2024-04-25 02-34-00](https://github.com/OpenRefine/OpenRefine/assets/158375948/a85a5a74-cb58-4df4-96e4-13cfe2fdaf9c)